backend/src/database/state-migrations/migrations.ts

```diff
@@ -1,31 +1,30 @@
-import type { UUID } from 'digital-fuesim-manv-shared';
+import type { StateExport, UUID } from 'digital-fuesim-manv-shared';
 import type { EntityManager } from 'typeorm';
-import type { ExerciseWrapper } from '../../exercise/exercise-wrapper';
-import { RestoreError } from '../../utils/restore-error';
+import { impossibleMigration } from './impossible-migration';
 
 /**
  * Such a function MUST update the initial state of the exercise with the provided {@link exerciseId} as well as every action associated with it from its current state version to the next version in a way that they are valid states/actions.
  * It MAY throw a {@link RestoreError} in a case where upgrading is impossible and a terminal incompatibility with older exercises is necessary.
  * It MUST update the respective updates to the exercise and its associated objects in the database.
  * All database interaction MUST use the provided {@link EntityManager}.
  */
-export type DbMigrationFunction = (
+export type DbMigration = (
     entityManager: EntityManager,
     exerciseId: UUID
 ) => Promise<void>;
 
 /**
- * Such a function MUST update the initial state of the provided {@link exerciseWrapper} as well as every action associated with it from its current state version to the next version in a way that they are valid states/actions.
+ * Such a function MUST update the current state, and, if present, the initial state and the action history of the provided {@link stateExport} from its current state version to the next version in a way that they are valid states/actions.
  * It MAY throw a {@link RestoreError} in a case where upgrading is impossible and a terminal incompatibility with older exercises is necessary.
- * It MUST NOT use the database.
+ * It MUST NOT use the database or any other resources like the exerciseMap.
  */
-export type InMemoryMigrationFunction = (
-    exerciseWrapper: ExerciseWrapper
-) => Promise<ExerciseWrapper>;
+export type StateExportMigration = (
+    stateExport: StateExport
+) => Promise<StateExport>;
 
-export interface MigrationFunctions {
-    database: DbMigrationFunction;
-    inMemory: InMemoryMigrationFunction;
+export interface Migrations {
```

Comment:
This gets confusing when having, e.g., this code:
```ts
const migrations: {
    [key: number]: Migrations;
};
```

Instead, I would just call this `Migration`. A `Migration` consists of a `DbMigration` and a `StateExportMigration`. If you want to, you could also call these `DbMigrationFunction` and  `StateExportMigrationFunction` to avoid the naming scheme being misunderstood to imply an inheritance relation.

Reply:
And yes - this is a contradiction to my comment before ^^